### ISSUE TITLE
Add a `Margin` modifier

### ISSUE THREAD

**Issue by zhiburt** [zhiburt/tabled#82]
> Margins
>
> Margins can be set outside the grid (top, bottom, left, right). Margins and the character used for rendering the margin can be set separately.
> 
> ```text
> vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
> >>>┌───────────┬───────────┐<<<<
> >>>│row 1 col 1│row 1 col 2│<<<<
> >>>├───────────┼───────────┤<<<<
> >>>│row 2 col 1│row 2 col 2│<<<<
> >>>└───────────┴───────────┘<<<<
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ```

There's a list of steps necessary to take.

- A new `margin` field must be added to `papergrid::Grid`.
- `fmt::Display` of `papergrid::Grid` must be updated.
- `margin.rs` with `Margin` modifier must be created in `tabled`  which will set the table margin.

The possible implementation of `tabled::Margin`

```rust
struct Margin {
  top: Option<Indent>,
  bottom: Option<Indent>,
  left: Option<Indent>,
  right: Option<Indent>,
}

struct Indent {
  size: usize,
  symbol: char,
}
```

_______

Inspired by the same functionality in https://github.com/vdmeer/asciitable

**Comment by kozmod** [comment]
I hope you don't mind that I'll get this one as the next task?

**Comment by zhiburt** [comment]
No way I could mind against this :)
You're welcome.

**Comment by zhiburt** [comment]
Hi @kozmod just realized 1 thing.

Maybe it's valuable to be able to set a `lead` side?

I mean what about the prior example I've provided.



> > ```
> > vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
> > >>>┌───────────┬───────────┐<<<<
> > >>>│row 1 col 1│row 1 col 2│<<<<
> > >>>├───────────┼───────────┤<<<<
> > >>>│row 2 col 1│row 2 col 2│<<<<
> > >>>└───────────┴───────────┘<<<<
> > ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> > ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> > ```

It could actually also look like the following.

```
>>>vvvvvvvvvvvvvvvvvvvvvvvvvvvvv
>>>┌───────────┬───────────┐<<<<
>>>│row 1 col 1│row 1 col 2│<<<<
>>>├───────────┼───────────┤<<<<
>>>│row 2 col 1│row 2 col 2│<<<<
>>>└───────────┴───────────┘<<<<
>>>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>>>^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Like left margin overrides a top and bottom, but not otherwise.
What do you think is it valuable?

**Comment by zhiburt** [comment]
A order could be checked by a set methods order.
If we would make Margin responsible only for 1 side per call.

Example

```rust
// mean that left overrides top
.with(Margin::top(3))
.with(Margin::left(3))


// mean that top overrides left && right
.with(Margin::left(3))
.with(Margin::right(3))
.with(Margin::top(3))
```

**Comment by zhiburt** [comment]
closed in #89
